packages/fxa-auth-server/lib/senders/email.js

```diff
@@ -481,98 +490,113 @@ module.exports = function (log, config) {
-      return Promise.all(
-        services.map((service) => {
-          const headers = {
-            'Content-Language': localized.language,
-            'X-Template-Name': template,
-            'X-Template-Version': templateVersion,
-            ...message.headers,
-            ...optionalHeader('X-Device-Id', message.deviceId),
-            ...optionalHeader('X-Flow-Id', message.flowId),
-            ...optionalHeader('X-Flow-Begin-Time', message.flowBeginTime),
-            ...optionalHeader('X-Service-Id', message.service),
-            ...optionalHeader('X-Uid', message.uid),
-          };
-
-          const { mailer, emailAddresses, emailService, emailSender } = service;
-
-          // Set headers that let us attribute success/failure correctly
-          message.emailService = headers['X-Email-Service'] = emailService;
-          message.emailSender = headers['X-Email-Sender'] = emailSender;
-
-          if (this.sesConfigurationSet && emailSender === 'ses') {
-            // Note on SES Event Publishing: The X-SES-CONFIGURATION-SET and
-            // X-SES-MESSAGE-TAGS email headers will be stripped by SES from the
-            // actual outgoing email messages.
-            headers[X_SES_CONFIGURATION_SET] = this.sesConfigurationSet;
-            headers[X_SES_MESSAGE_TAGS] = sesMessageTagsHeaderValue(
-              message.metricsTemplate || template,
-              emailService
-            );
-          }
+    const services = await this.selectEmailServices(message);
 
-          log.debug('mailer.send', {
-            email: emailAddresses[0],
-            template,
-            headers: Object.keys(headers).join(','),
-          });
+    for (const service of services) {
+      const headers = {
+        'Content-Language': localized.language,
+        'X-Template-Name': template,
+        'X-Template-Version': templateVersion,
+        ...message.headers,
+        ...optionalHeader('X-Device-Id', message.deviceId),
+        ...optionalHeader('X-Flow-Id', message.flowId),
+        ...optionalHeader('X-Flow-Begin-Time', message.flowBeginTime),
+        ...optionalHeader('X-Service-Id', message.service),
+        ...optionalHeader('X-Uid', message.uid),
+      };
 
-          const emailConfig = {
-            sender: this.sender,
-            from: this.sender,
-            to: emailAddresses[0],
-            subject: localized.subject,
-            text: localized.text,
-            html: localized.html,
-            xMailer: false,
-            headers,
-          };
+      const { mailer, emailAddresses, emailService, emailSender } = service;
+      const to = emailAddresses[0];
+
+      try {
+        await bounces.check(to);
+      } catch (err) {
+        log.error('mailer.send.error', {
+          err: err.message,
+          code: err.code,
+          errno: err.errno,
+          to,
+          emailSender,
+          emailService,
+          template,
+        });
```

Comment:
This is copied from below in the `sendMail` callback, and is related to @biancadanforth's PR #10543 ... but I think it'd be better as it's own distinct log entry. I'm noting it here so that if we do change it we'll know about it for monitoring.

Reply:
What do you mean by making it its own distinct log entry?

Reply:
as in with it's own "name" that we can query like `email.bounce` instead of `mailer.send.error`